include/envoy/stream_info/stream_info.h

```diff
@@ -202,6 +202,15 @@ class StreamInfo {
    */
   virtual void onUpstreamHostSelected(Upstream::HostDescriptionConstSharedPtr host) PURE;
 
+  /**
+   * @param std::string name denotes the name of the route.
+   */
+  virtual void setRouteName(std::string name) PURE;
```

Comment:
Can you make this take an absl::string_view instead? that would be more flexible.